VDA-DID-Registry/contracts/VeridaDIDRegistry.sol

```diff
@@ -5,419 +5,163 @@ pragma solidity ^0.8.6;
 import { VeridaDataVerificationLib } from "./VeridaDataVerificationLib.sol";
 import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 
-/** @title VeridaDIDRegistry */
-contract VeridaDIDRegistry is OwnableUpgradeable {
-
-  struct DelegateParam {
-    bytes32 delegateType;
-    address delegate;
-    uint validity;   
-  }
-
-  struct RevokeDelegateParam {
-    bytes32 delegateType;
-    address delegate;
-  }
-
-  struct AttributeParam{
-    bytes32 name;
-    bytes value;
-    uint validity;
-    bytes proof;
-  }
-
-  struct RevokeAttributeParam{
-    bytes32 name;
-    bytes value;
-  }
+import "./IVeridaDIDRegistry.sol";
+import "./EnumerableSet.sol";
 
-  mapping(address => address) public owners;
-  mapping(address => mapping(bytes32 => mapping(address => uint))) private delegates;
-  mapping(address => uint) public changed;
-  mapping(address => uint) private nonce;
-
-  event DIDOwnerChanged(
-    address indexed identity,
-    address owner,
-    uint previousChange
-  );
-
-  event DIDDelegateChanged(
-    address indexed identity,
-    bytes32 delegateType,
-    address delegate,
-    uint validTo,
-    uint previousChange
-  );
+/** @title VeridaDIDRegistry */
+contract VeridaDIDRegistry is OwnableUpgradeable, IVeridaDIDRegistry {
 
-  event DIDAttributeChanged(
-    address indexed identity,
-    bytes32 name,
-    bytes value,
-    uint validTo,
-    bytes proof,
-    uint previousChange
-  );
+  using EnumerableSet for EnumerableSet.StringSet;
 
   /**
-   * @notice Initialize
+   * @notice Map of controllers
+   * @dev DID address => controller address
    */
-  function initialize() public initializer {
-    __Ownable_init();
-  }
+  mapping (address => address) private _controllers;
 
   /**
-   * Get current nonce of DID
-   *@param did - DID registered here
+   * @notice Flags for registered status
+   * @dev DID address => bool
    */
-  function getNonce(address did) public view returns(uint) {
-    return nonce[did];
-  }
+  mapping (address => bool) private _isRegistered;
 
   /**
-   * @notice Return owner of DID
-   * @param identity - DID
-   * @return - owner address of DID
+   * @notice Map of nonce
+   * @dev DID address => nonce
    */
-  function identityOwner(address identity) public view returns(address) {
-     address owner = owners[identity];
-     if (owner != address(0x00)) {
-       return owner;
-     }
-     return identity;
-  }
+  mapping(address => uint) private _nonce;
```

Comment:
Add a comment stating this is the "next" nonce, not the current.

Reply:
Updated.